interfaces/cython/cantera/composite.py

```diff
@@ -1306,11 +1308,45 @@ def restore(self, fname, name=None, sub=None):
         self.shape = self._api_shape()
         return meta
 
+    def _to_picklable(self):
+        with _NamedTemporaryFile(suffix=".yaml", delete=False) as t_file:
```

Comment:
I don't think this needs to be a context manager, I think it can be a regular assignment


```suggestion
        t_file = _NamedTemporaryFile(suffix=".yaml", delete=False)
```

Alternatively, you can set `delete_on_close=False, delete=True` and put all usage of the file inside the context manager. Then `os.unlink` is not needed.

Reply:
Definitely not necessary - it was a leftover from earlier tests. For the context manager, there's a race condition on Windows - the process doesn't want to share the file, which is avoided by the `delete=False` option.

Reply:
My read of the docs is that setting `delete_on_close=False` allows the context manager to be used even on Windows, as long as any additional uses of the file are closed before exiting the temp file context manager. 

> delete is true but delete_on_close is false. Note, that in this case the additional opens that do not share delete access (e.g. created via builtin open()) must be closed before exiting the context manager, else the os.unlink() call on context manager exit will fail with a PermissionError.

Regardless, this solution works just as well! 😊